include/console.h

```diff
@@ -0,0 +1,36 @@
+#ifndef _SYS_CONSOLE_H_
+#define _SYS_CONSOLE_H_
+
+#include <linker_set.h>
+
+typedef struct console console_t;
+
+typedef void (*cn_init_t)(console_t *);
+typedef int (*cn_getc_t)(console_t *);
+typedef void (*cn_putc_t)(console_t *, int);
+
+typedef struct console {
+  cn_init_t cn_init;
+  cn_getc_t cn_getc;
+  cn_putc_t cn_putc;
+  int cn_prio;
+} console_t;
+
+#define CONSOLE_DEFINE(name, prio)                                             \
```

Comment:
Right, I did not take into account code browsability factor. It's good you pointed it out, as I started using this construct in consequent changes.